tinygrad/function.py

```diff
@@ -36,13 +37,19 @@ def forward(self, x:LazyBuffer) -> LazyBuffer:
   def backward(self, grad_output:LazyBuffer) -> LazyBuffer:
     return grad_output.e(UnaryOps.NEG).e(BinaryOps.MUL, self.ret).e(BinaryOps.MUL, self.ret)
 
+# ************* unary ops *************
 class Sin(Function):
-  def forward(self, x:LazyBuffer) -> LazyBuffer:
+  def forward(self, x: LazyBuffer) -> LazyBuffer:
     self.x = x
+    self.fast_approx = is_dtype_fastmath_supported(x.dtype)
+    if self.fast_approx:
+      return xsin(x)
     return x.e(UnaryOps.SIN)
 
-  def backward(self, grad_output:LazyBuffer) -> LazyBuffer:
-    return self.x.const(math.pi / 2).e(BinaryOps.ADD, self.x.e(UnaryOps.NEG)).e(UnaryOps.SIN).e(BinaryOps.MUL, grad_output)
+  def backward(self, grad_output: LazyBuffer) -> LazyBuffer:
```

Comment:
Can you remove the space between `param` and `Type`? This is what we follow across the codebase:

```diff
- def backward(self, grad_output: LazyBuffer) -> LazyBuffer:
+ def backward(self, grad_output:LazyBuffer) -> LazyBuffer:
```